### DISCUSSION TITLE
Changing 'size' prop of  ReactiveList causes the total number of hits to reset to 10000

### DISCUSSION THREAD

**Discussion by malsioufi** [appbaseio/reactivesearch#1563] [Q&A] [answered]
**Affected Projects**
React

**Library Version:** x.y.z
^3.12.7

**Describe the bug**
I am using ReactiveList component to show the results of my search.
I have added to the application a select box where the user can choose number of results to be shown per page. so I am setting the size dynamically. The functionality works fine and the component updates correctly, however, I noticed that the number of total hits by default is for example 2 millions 

but when I change the number of pages it drops down to 10000.

**To Reproduce**

1. Generate big amount of data, e.g. 2 million documents.
2. Add component that control number of elements to be shown in page ( some state like state.numberOfElements)
3. Use ReactiveList component with pagination=true, size={state.numberOfElements}
4. Change number of elements

**Expected behavior**

The page show number of elements equals to what is set and keep showing number of total hits as before in this example 2 million 

**Screenshots**

Before changing number of elements
![image](https://user-images.githubusercontent.com/5476902/100447926-be996280-30b1-11eb-84f8-0a68204fd917.png)
Afterchanging number of elements
![image](https://user-images.githubusercontent.com/5476902/100447960-d1139c00-30b1-11eb-9df0-aa441a002afc.png)
**Desktop (please complete the following information):**

- OS: Windos 10
- Browser Google Chrome
- Version 86.0.4240.198

**Additional context**

I have previously had an issue when I was not able to get more than 10K hits and it was solved 
https://github.com/appbaseio/reactivesearch/issues/1558

**Comment by KalleVuorjoki** [comment, marked as answer]
This is default Elasticsearch 7.x [setting](https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#track-total-hits-10000-default). Since Elasticsearch 6.8.x is maintained until 8.x is released, it might not useful to resolve this issue on application level. 
You can get the total results by adding `?rest_total_hits_as_int=true` in your request parameter.

**Reply by malsioufi** [reply]
Thanks, that sounds reasonable! But where can I set this parameter?

**Comment by allendol** [comment]
I have the same problem . I set defaultQuery as below
   defaultQuery={props => ({
         track_total_hits: true 
    })}		
Now I get the correct total hits more than default 10000,
But when the 'size' property is changed defaultQuery doesn't work and total hits goes back 10000.
I am not sure how and where I can pass ?rest_total_hits_as_int=true in your request parameter.
Any sample code is appreciated.

**Reply by SavvyShah** [reply]
The issue cannot be reproduced.
I have made a [codesandbox](https://ow4mez.csb.app/) where you can try changing the size by hitting the button. You would see that the size of the list gets bigger and the total results (26444) would remain the same.
The loom preview for demo is attached below. It shows the network requests which still track the total hits and doesn't get reset to default when changing size.

[Loom preview](https://www.loom.com/share/bb7ef08b037d4fbb89aceb31b78ea0c8)
[CSB Link](https://codesandbox.io/s/ow4mez)
